src/test/java/fakes/UserValidatorTest.java

```diff
@@ -1,49 +1,315 @@
 package fakes;
 
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.mockito.Mockito.mock;
+
 class UserValidatorTest {
 
     @Nested
     class isValidUsername{
 
         @Test
         void returnsTrueIfOnlyLetters(){
-            throw new IllegalArgumentException("you should implement code here");
+            //kein mokito
```

Comment:
Die Idee all dieser isValidUsername Testfälle ist es, sicherzustellen, dass alle Szenarien (z.B. "OnlyLetters") von der Methode isValidUsername korrekt erkannt werden.
z.B. für den `returnsTrueIfOnlyLetters` Testfall könnten die Assertion sein:  `Assertions.assertFalse(new User(",,,,").isValidUsername()); Assertions.assertTrue(new User("abcdefg").isValidUsername());`

Im Testfall selbst befindet sich keine if oder for Schlaufe. Dieser Validierungsalgorithmus muss in die User Klasse.